src/riscv/lib/src/machine_state/memory/state.rs

```diff
@@ -227,18 +226,25 @@ where
         address: Address,
         length: usize,
         perms: Permissions,
+        memory_governance_listener: &mut impl MemoryGovernanceListener,
     ) -> Result<(), super::MemoryGovernanceError>
     where
         M: ManagerWrite,
     {
         Self::check_bounds(address, length, super::MemoryGovernanceError)?;
 
-        self.readable_pages
-            .modify_access(address, length, perms.can_read());
-        self.writable_pages
-            .modify_access(address, length, perms.can_write());
-        self.executable_pages
-            .modify_access(address, length, perms.can_exec());
+        if let Some(length) = NonZeroUsize::new(length) {
```

Comment:
If length is zero, you don't need to do anything, right? Might as well exit early if it is `None` via `let Some(length) = ... else { return Ok(()); }` at the start of the function. Or alternatively, this function could accept non-zero lengths too.